docs/tutorials/04_stream/viz_interaction.py

```diff
@@ -0,0 +1,135 @@
+"""
+====================================
+Streaming FURY with user interaction
+====================================
+
```

Comment:
the tutorial needs more explanation

Reply:
> Hi @devmessias,
> 
> Thank you for this excellent new module. Here are some of my issues:
> 
> * The documentation generation does not work anymore.
> * it would be good to provide more textual information/description in each tutorial
> * There are no tests. Too much code not tested.
> * Many functions are not documented.
> 
> See below for some other comments.
> 
> Please, can you do a sprint on this issue. thank you.
> @filipinascimento, Can you help @devmessias to finalize this module. Thank you

I'll try to cover more cases inside of widget.py and server.py. However, I couldn't find a good way to cover some specific parts of the streaming system

### 1 Optional modules and python version
![image](https://user-images.githubusercontent.com/6979335/127545661-44ba83ad-7b4e-4aea-8f03-e94af7255627.png)

opencv and Cython are not mandatory. Because these libs just provides a better performance for some specific tasks. However, this approach has an issue: pytest will cover the methods based on the installed packages.

For example, this part of the code.
![image](https://user-images.githubusercontent.com/6979335/127546216-961a99c9-4d41-47e9-8d88-46ba8d738019.png)

Maybe this is a solution: https://coverage.readthedocs.io/en/latest/cmd.html#combining-data-files-coverage-combine
However we need to change the github action to perform multiple tests with different requirements (and python versions) and then use *coverage combine* to join all the reports

### 2 aiohttp endpoints and webserver
I need to discuss that with @filipinascimento
![image](https://user-images.githubusercontent.com/6979335/127547537-c2f6147f-eae6-46e5-b946-eba0188461a0.png)

We have a problem because the usual flow for the streaming system is that the aiohttp server need to be created in a unique process. Therefore, we will not capture anything running pytest. In the first version of #437  was possible to create the webserver in the main thread. However, @filipinascimento and I have agreed to remove this behavior. Maybe we need to get this behavior back.

### 3 Python core bug
https://bugs.python.org/issue38119 
![image](https://user-images.githubusercontent.com/6979335/127548939-e6051125-1dd5-41d7-afeb-9617dfd6461b.png)

![image](https://user-images.githubusercontent.com/6979335/127549057-e80462f5-fb6f-4e99-9a4a-3019bb5a55b2.png)

This is more problematic. This `try exception` was not caught by pytest because the exception is called only if something goes wrong with the  semaphores and shared memory. Maybe I can force a memory error to test this. But I don't know if it's a good idea. I can also add  `# pragma: no cover` to  avoid those lines being added in the coverage report.

For example,  I've used  `# pragma: no cover`  inside of the abstractmethods 
![image](https://user-images.githubusercontent.com/6979335/127549831-5bf5570d-794b-4f0a-9f67-19e565fdf5f1.png)

Reply:
If we run this tutorial without aiortc installed, it runs but the link displayed is for webrtc and not mjpeg. If webrtc is not available the link should point out to mjpeg encoding instead.

Reply:
Maybe we need a section in the documentation describing better the architecture and usage of the streaming system. A glossary of terms (WebRTC, h264, mjpeg, etc etc) and how the different optional packages improve performance.